Geometry_Engine/Compute/CurveProximity.cs

```diff
@@ -0,0 +1,1004 @@
+                result4.Item1 = curve2.ClosestPoint(result4.Item2);
+                result4.Item2 = curve1.ClosestPoint(result4.Item1);
+            } while (oldresult.Item2.Distance(result4.Item2) > tolerance * tolerance && oldresult.Item1.Distance(result4.Item1) > tolerance * tolerance);
+                result5.Item2 = curve2.PointAtParameter(0.5);
+                result5.Item1 = curve1.ClosestPoint(result5.Item2);
+                result6.Item1 = curve1.PointAtParameter(0.5);
+                result6.Item2 = curve2.ClosestPoint(result6.Item1);
+            do
+            {
+                oldresult.Item1 = result5.Item1.Clone();
+                oldresult.Item2 = result5.Item2.Clone();
+                result5.Item1 = curve2.ClosestPoint(result5.Item2);
+                result5.Item2 = curve1.ClosestPoint(result5.Item1);
+            } while (oldresult.Item2.Distance(result5.Item2) > tolerance * tolerance && oldresult.Item1.Distance(result5.Item1) > tolerance * tolerance);
+            do
+            {
+                oldresult.Item1 = result6.Item1.Clone();
+                oldresult.Item2 = result6.Item2.Clone();
+                result6.Item1 = curve2.ClosestPoint(result6.Item2);
+                result6.Item2 = curve1.ClosestPoint(result6.Item1);
+            } while (oldresult.Item2.Distance(result6.Item2) > tolerance * tolerance && oldresult.Item1.Distance(result6.Item1) > tolerance * tolerance);
+            if (result.Item1 == null || result.Item2 == null)
+                result = result2;
+            if (result2.Item2.Distance(result2.Item1) < result.Item1.Distance(result.Item2))
+                result = result2;
+            if (result3.Item2.Distance(result3.Item1) < result.Item1.Distance(result.Item2))
+                result = result3;
+            if (result4.Item2.Distance(result4.Item1) < result.Item1.Distance(result.Item2))
+                result = result4;
+            if (result5.Item2.Distance(result5.Item1) < result.Item1.Distance(result.Item2))
+                result = result5;
+            if (result6.Item2.Distance(result6.Item1) < result.Item1.Distance(result.Item2))
+                result = result6;
+            return result;
+        }
+
+        /***************************************************/
+
+        public static BH.oM.Reflection.Output<Point, Point> CurveProximity(this Arc curve1, Circle curve2, double tolerance = Tolerance.Distance)
+        {
+            if (curve1.CurveIntersections(curve2).Count > 0)
+                return new BH.oM.Reflection.Output<Point, Point> { Item1 = curve1.CurveIntersections(curve2)[0], Item2 = curve1.CurveIntersections(curve2)[1] };
+            BH.oM.Reflection.Output<Point, Point> result = new BH.oM.Reflection.Output<Point, Point>();
+            Plane ftPln1 = curve1.FitPlane();
+            Plane ftPln2 = curve2.FitPlane();
+            List<Point> intPts = new List<Point>();
+            Point tmp = new Point();
```

Comment:
Why is this Point instantiated here?

Reply:
in 2 places it needs to be instantiated because of later comparisons/checks. In other places though, i removed the allocation.

Reply:
In this case it should be null, instantiating (0,0,0) is risky as it might meet some `if` statement (edge case). Wouldn't that be possible?